### DISCUSSION TITLE
feat: print full stack trace on error

### DISCUSSION THREAD

**Discussion by tmillr** [avajs/ava#3144] [Q&A]
* What you're trying to do
**Debug my code.**

* Why you can't use AVA for this
**No stack trace on error? Only error message is printed.**

* And maybe how you think AVA could handle this
**Print full stack trace on error so that I can find out where they are occurring, including when they occur within async fn's. Or, at the very least make it opt-in.**

Things can get really painful when errors occur and there's only an error message but no stack trace. I'm not sure if this is something specific to async functions and/or specific to AVA, but it can get pretty annoying when rapidly prototyping with vanilla JS. Is this the expected behavior? Or maybe I've messed something up?

<img width="954" alt="Screen Shot 2022-11-26 at 4 46 17 AM" src="https://user-images.githubusercontent.com/45028928/204089680-2e0fc0e3-0544-41ee-b782-41b5ea6dabbd.png">

**Comment by novemberborn** [comment]
It's hard to say, there is a stack trace but it starts at the test line.

Which AVA version are you using? Are you able to share a reproduction?

**Reply by tmillr** [reply]
v5.0.1, standard settings (no config), vanilla js

```js
const hasFullStackTraceOnlyOutsideNotThrowsAsync = async () => {
  throw new Error("some error");
};

test("throwing async fn, returned promise handled by t.notThrowsAsync(): missing full trace", async (t) => {
  await t.notThrowsAsync(hasFullStackTraceOnlyOutsideNotThrowsAsync());
});
```

I believe the full trace comes back if you either:

A. remove `async` from `hasFullStackTraceOnlyOutsideNotThrowsAsync` definition to make it sync

or

B. remove `t.notThrowsAsync()` assertion altogether leaving only the other function call (and there will be a full trace no matter if that call is awaited or not)

**Reply by novemberborn** [reply]
Sorry, this got caught up in pre-holiday busyness and then holiday-without-a-keyboardness. I'll have to play with this… it's odd because either code path should process the error the same way, but maybe within the assertion the async-stuff gets tripped up a bit at a JS level.
